commit eb54943ae270becd065e3b15eacbd3cd7c2ca0e2 (refs/changes/07/4607/4)
Author: Chris Burroughs <chris.burroughs@joyent.com>
Date:   2018-08-24T14:27:21+00:00 (1 year, 1 month ago)
    
    TRITON-649 cnapi sendTaskRequest fails to read cn-agent port
    Reviewed by: Josh Wilsdon <josh@wilsdon.ca>
    Approved by: Josh Wilsdon <josh@wilsdon.ca>

diff --git a/lib/models/server.js b/lib/models/server.js
index 6c2215e..7267877 100644
--- a/lib/models/server.js
+++ b/lib/models/server.js
@@ -17,7 +17,6 @@
 var async = require('async');
 var assert = require('assert-plus');
 var dns = require('dns');
-var jsprim = require('jsprim');
 var qs = require('querystring');
 var restify = require('restify');
 var sprintf = require('sprintf').sprintf;
@@ -1879,8 +1878,9 @@ function (opts) {
             }
 
             if (sysinfo.hasOwnProperty('CN Agent Port')) {
-                if (!jsprim.parseInteger('CN Agent Port') instanceof Error) {
-                    serverAdminPort = sysinfo['CN Agent Port'];
+                const port = Number.parseInt(sysinfo['CN Agent Port']);
+                if (!Number.isNaN(port)) {
+                    serverAdminPort = port;
                 }
             }
 
@@ -2087,8 +2087,9 @@ ModelServer.prototype.sendRequest = function (opts, cb) {
             }
 
             if (sysinfo.hasOwnProperty('CN Agent Port')) {
-                if (!jsprim.parseInteger('CN Agent Port') instanceof Error) {
-                    serverAdminPort = sysinfo['CN Agent Port'];
+                const port = Number.parseInt(sysinfo['CN Agent Port']);
+                if (!Number.isNaN(port)) {
+                    serverAdminPort = port;
                 }
             }
 
